sklearn/_isotonic.pyx

```diff
@@ -69,3 +69,50 @@ def _isotonic_regression(np.ndarray[DOUBLE, ndim=1] y,
             break
 
     return solution
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def _make_unique(np.ndarray[dtype=np.float64_t] X,
+                  np.ndarray[dtype=np.float64_t] y,
+                  np.ndarray[dtype=np.float64_t] sample_weights):
+    # assumes ordered X
```

Comment:
There is an inline comment here, but I should probably write a proper docstring. Wil ldo that now.